pallets/msa/src/lib.rs

```diff
@@ -1383,39 +1393,59 @@ impl<T: Config> Pallet<T> {
 				Error::<T>::SignatureRegistryLimitExceeded
 			);
 
-			// Move the oldest in the ring to the next oldest signature
+			// Move the oldest in the list to the next oldest signature
 			oldest = next_oldest.clone();
 
-			<PayloadSignatureRegistryRing<T>>::remove(pointer.oldest);
+			<PayloadSignatureRegistryList<T>>::remove(pointer.oldest);
 		}
 
-		// Update the prior newest
-		// We could remove this > 0, but then we'd have to be OK with the None not failing
-		if pointer.count > 0 {
-			let mut prior_newest = Self::get_payload_signature_registry(pointer.newest.clone())
-				.ok_or(Error::<T>::SignatureRegistryCorrupted)?;
-
-			prior_newest.1 = signature.clone();
-
-			<PayloadSignatureRegistryRing<T>>::insert(pointer.newest, prior_newest);
+		// Add the newest signature if we are not the first
+		if pointer.count != 0 {
+			<PayloadSignatureRegistryList<T>>::insert(
+				pointer.newest,
+				(pointer.newest_expires_at, signature.clone()),
+			);
 		}
 
-		// Add the newest signature to complete the ring
-		<PayloadSignatureRegistryRing<T>>::insert(
-			signature,
-			(signature_expires_at, oldest.clone()),
-		);
-
-		PayloadSignatureRegistryRingPointer::<T>::put(SignatureRegistryPointer {
-			// The count doesn't change if buffer is full
-			count: if is_ring_buffer_full { pointer.count } else { pointer.count + 1 },
+		// Update the pointer.newest to have the signature that just came in
+		PayloadSignatureRegistryPointer::<T>::put(SignatureRegistryPointer {
+			// The count doesn't change if list is full
+			count: if is_registry_full { pointer.count } else { pointer.count + 1 },
 			newest: signature.clone(),
+			newest_expires_at: signature_expires_at,
 			oldest,
 		});
 
 		Ok(())
 	}
 
+	/// Used to fill up the registry for benchmarks
+	#[cfg(feature = "runtime-benchmarks")]
```

Comment:
I think we can move this too the benchmark file.

Reply:
Yes. Yes we can. I didn't realize I could call the storage directly.

Moving...